program-runtime/src/loaded_programs.rs

```diff
@@ -318,13 +322,51 @@ impl LoadedPrograms {
 
     /// Evicts programs which were used infrequently
     pub fn sort_and_evict(&mut self) {
-        // TODO: Sort programs by their usage_counter
-        // TODO: Truncate the end of the list
+        // Sort programs by their usage_counter wile keeping track on the
+        // program key and the slot.
+        let mut sorted_programs: Vec<(&Arc<LoadedProgram>, &Pubkey, usize)> = self.entries
+            .iter()
+            .flat_map(|(x, y)| y.iter().enumerate().map(move |z| (z.1, x, z.0)))
```

Comment:
let's use slightly more descriptive names for the variables, instead of x, y, z etc. The code will be more readable.